contracts/bonding/BondingManager.sol

```diff
@@ -1263,72 +1297,47 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         uint256 currentBondedAmount = del.bondedAmount;
         uint256 currentFees = del.fees;
 
-        uint256 LIP_36_ROUND = roundsManager().LIPUpgradeRound(36);
+        uint256 lip36Round = roundsManager().lipUpgradeRound(36);
 
         // Only will have earnings to claim if you have a delegate
         // If not delegated, skip the earnings claim process
         if (del.delegateAddress != address(0)) {
-            bool isTranscoder = _delegator == del.delegateAddress;
-
-            Transcoder storage transcoder = transcoders[del.delegateAddress];
-            if (startRound <= LIP_36_ROUND) {
+            if (startRound <= lip36Round) {
                 // Cannot claim earnings for more than maxEarningsClaimsRounds before LIP-36
                 // This is a number to cause transactions to fail early if
                 // we know they will require too much gas to loop through all the necessary rounds to claim earnings
                 // The user should instead manually invoke `claimEarnings` to split up the claiming process
                 // across multiple transactions
-                uint256 endLoopRound = _endRound <= LIP_36_ROUND ? _endRound : LIP_36_ROUND;
+                uint256 endLoopRound = _endRound <= lip36Round ? _endRound : lip36Round;
                 require(endLoopRound.sub(_lastClaimRound) <= maxEarningsClaimsRounds, "too many rounds to claim through");
-
-                while (startRound <= endLoopRound) {
-                    EarningsPool.Data storage earningsPool = transcoder.earningsPoolPerRound[startRound];
-
-                    if (startRound == LIP_36_ROUND && !earningsPool.hasTranscoderRewardFeePool) {
-                        break;
-                    }
-
-                    if (earningsPool.hasClaimableShares()) {
-                        (uint256 fees, uint256 rewards) = earningsPool.claimShare(currentBondedAmount, isTranscoder);
-                        currentFees = currentFees.add(fees);
-                        currentBondedAmount = currentBondedAmount.add(rewards);
-                    }
-
-                    startRound = startRound.add(1);
-                }
             }
 
+            (
+                currentBondedAmount,
+                currentFees
+            ) = pendingStakeAndFees(_delegator, _endRound);
+
             // Check whether the endEarningsPool is initialised
             // If it is not initialised set it's cumulative earnings factors (fees and rewards)
             // This serves two purposes:
             // - Initialise the correct values for the current claimEarnings 'epoch' to calculate cumulative earnings
```

Comment:
I believe this is no longer correct, it only serves the single purpose of initialising the correct `lastClaimRound` earningsPool for the next "epoch" (for lack of better terminology)

Reply:
Fixed in 654de5b